src/Emu/Metadata/WildlifeAcoustics/WamdExtractor.cs

```diff
@@ -31,46 +32,72 @@ public ValueTask<Recording> ProcessFileAsync(TargetInformation information, Reco
         {
             var stream = information.FileStream;
 
-            var tryWamdData = Wamd.ExtractMetadata(stream);
+            var tryWamdData = WamdParser.ExtractMetadata(stream);
 
             if (tryWamdData.IsSucc)
             {
                 Wamd wamdData = (Wamd)tryWamdData;
 
-                int numMicrophones = wamdData.MicrophoneType.Length;
+                int numMicrophones = wamdData.MicType.Length;
 
                 var location = recording.Location;
-                if (wamdData.Location is not null)
+                if (wamdData.GpsFirst is not null)
                 {
-                    location = wamdData.Location;
+                    location = wamdData.GpsFirst;
+                }
+                else if (wamdData.PosLast is not null)
+                {
+                    location = wamdData.PosLast;
+                }
+
+                if (wamdData.FileStartTime is not null)
+                {
+                    if (wamdData.FileStartTime.Value.Case is OffsetDateTime o)
```

Comment:
Possible rename `o` to a more descriptive name?

I had to read the proceeding lines of code to figure out how `o` was used

Reply:
hmmm this is a common convention for temporary variables that created just due to a cast. Here `o` it is just the cast value of `wamdData.FileStartTime`. However, any example of something difficult to read takes priority